### PR TITLE
Make the FTL log tail optional. Pass in `TAIL_FTL_LOG: 0` to silence it

### PR DESCRIPTION
**Pull request by PromoFaux**
### **What does this PR aim to accomplish?:**

See title. By default the image will tail the FTL log. Users can disable this if it is too noisy for their preference

---
**By submitting this pull request, I confirm the following:** 

1. I have read and understood the [contributors guide](https://docs.pi-hole.net/guides/github/contributing/), as well as this entire template. I understand which branch to base my commits and Pull Requests against. 
2. I have commented my proposed changes within the code and I have tested my changes.
3. I am willing to help maintain this change if there are issues with it later.
4. It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
5. I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))

---
- [x] I have read the above and my PR is ready for review. _Check this box to confirm_

### CODE REVIEW

**Review by edgd1er — COMMENTED** [review]
If no logs are required may be pihole-FTL could be started et kept in foreground ('-f)

**Comment by PromoFaux** [comment]
If we did that, then the container would stop if ever FTL was restarted/stopped.

For now at least I'd like to keep the command as-is

**Comment by edgd1er** [comment]
Could https://github.com/Yelp/dumb-init be of any help to handle Init 1 pid ? it would send signals (3,9,15) to pihole-FTL requiring a proper process termination and database would stay clean. All process started with the script would be properly terminated also:
(tail, logrotate if any, crond ....)

"Normally, a signal like SIGTERM received by a shell isn't forwarded to subprocesses; instead, only the shell process dies. With dumb-init, you can just write shell scripts with dumb-init in the shebang"

**Comment by PromoFaux** [comment]
Tempted to go back to s6, tbh.

**Comment by Gontier-Julien** [comment]
> Tempted to go back to s6, tbh.
> 
> Edit: That said, something like `dumb-init` or `tini` is probably enough these days.
> 
> `tini` supports more architectures by the looks of things (just going by releases)

One other advantage of `tini` is that it is supported by docker.

From they're readme page:

> NOTE: If you are using Docker 1.13 or greater, Tini is included in Docker itself. This includes all versions of Docker CE. To enable Tini, just [pass the --init flag to docker run](https://docs.docker.com/engine/reference/commandline/run/).

And it even in the alpine repo too.

Maybe you did already know it, but i just wanted to point it out because it neat ^^
i think it would be nice if it is need to use `tini` over `s6` since it also lighter and enough as you said.

**Comment by PromoFaux** [comment]
Yeah, @dschaper already pointed out `tini` to me a while back, but I was hoping (rather naively) that we wouldn't need it, hence getting excited and ripping s6 out when I started on this version of the container!

**Comment by PromoFaux** [comment]
Converted to draft for the time being

**Comment by PromoFaux** [comment]
> If no logs are required may be pihole-FTL could be started et kept in foreground ('-f)

Something else I realised in doing some testing this evening. I am not sure what it is, but I remember it happening on v5 container too, if we start pihole-FTL in the foreground, it stops outputting to the docker log. Note there are more lines in the actual file than there are on the bottom half of the screen:

image

**Comment by DL6ER** [comment]
What is actually puzzling here is that it stops right in the middle of reporting the database statistics. They are simply a collection of `printf` statements, there is nothing in the code that could actually make it stop after the first line:

https://github.com/pi-hole/FTL/blob/5753b50df40defa15d16f6989e12a9ecf3439be9/src/log.c#L502-L512

**Comment by PromoFaux** [comment]
Marked conversation about init systems as outdated due to #1406 being merged. 

Starting FTL in the foreground leads to weirdness with the output where it just cuts out - see previous two comments. This works well enough. Either we tail the log or we don't. Simples.

**Review by yubiuser — CHANGES_REQUESTED** [review]
Needs documentation of the new env setting.

**Comment by PromoFaux** [comment]
@yubiuser  https://github.com/pi-hole/docker-pi-hole/pull/1429

File: src/start.sh [hunk 1]
```diff
@@ -126,7 +126,11 @@ start() {
   sh /opt/pihole/pihole-FTL-prestart.sh
   capsh --user=$DNSMASQ_USER --keep=1 -- -c "/usr/bin/pihole-FTL $FTL_CMD >/dev/null" &
 
-  tail -f /var/log/pihole-FTL.log &
+  if [ "${TAIL_FTL_LOG:-1}" -eq 1 ]; then
+    tail -f /var/log/pihole-FTL.log &
```
Comment (yubiuser) [inline src/start.sh:130]: You might want to change to

```suggestion
    tail -f /var/log/pihole/FTL.log &
```

As the other is only a simlink we wanted to remove with v6. See

https://github.com/pi-hole/pi-hole/blob/fef2861eae23442ad823d51c9b4bfdd5b38ab200/advanced/Templates/pihole-FTL-prestart.sh#L29-L34

Comment (PromoFaux) [reply src/start.sh]: Good point well made

**Review by yubiuser — APPROVED** [review]

**Pull request merged**